demo/mindstorms_widgets_demo.py

```diff
@@ -0,0 +1,255 @@
+        print( 'remote channel 1 key = ' + str(val) )
+        time.sleep(1)
+    wait_for_back_button()
+
+def run_med_motor( regulate ):
+    robot.motor_rotation_measure( 'medium', 'reset' )
+    for s in range (10,101,10):
+        robot.medium_motor( 'on', power=s, regulated=regulate )
+        time.sleep(1)
+        rotations=robot.motor_rotation_measure( 'medium', 'rotations' )
+        degrees=robot.motor_rotation_measure( 'medium', 'degrees' )
+        (speed_sp,duty_cycle_sp) = \
+            robot.motor_rotation_measure( 'medium', 'current_power' )
+        # speed_sp is in degrees/sec, duty_cycle_sp is total power
+        print( 'power=%s, duty_cycle=%s, deg/sec=%s, rotations=%s, '
+               'degrees=%s' %( str(s), str(duty_cycle_sp), str(speed_sp),
+                               str(rotations), str(degrees)) )
+    robot.medium_motor( 'off' )
+    
+def demo_motor_rotation():
+    draw_header( 'Motor Rotation' )
+    brick_print( '<enter> to start' )
+    wait_for_enter()
+    brick_print( 'Medium motor, regulated speed' )
+    print( 'Medium motor, regulated speed' )
+    run_med_motor( True )
+    brick_print( 'Medium motor, unregulated' )
+    print( 'Medium motor, unregulated speed' )
+    run_med_motor( False )
+    wait_for_back_button()
+
+def draw_header( header ):
+    lcd.clear()
+    lcd.draw.text( (2,20), header )
+    lcd.draw.line( (0, 32, hpix, 32), width=2 )
+    lcd.y_pos = 40
+    
+def brick_print( line, fill='black', update=True ):
+    lcd.draw.text( (2,lcd.y_pos), line, fill=fill )
+    lcd.y_pos += 8
+    if update:
+        lcd.update()
+
+def draw_menu( selection ):
+    draw_header( 'Mindstorm Widgets (Back=exit)' )
+    for i in range( 0, len(menu) ):
+        if i == selection:
+            lcd.draw.rectangle( [0, lcd.y_pos, hpix, lcd.y_pos+9], 
+                                fill="black" )
+            brick_print( '>' + menu[selection][1], fill='white',
+                            update=False )
+        else:
+            brick_print( ' ' + menu[i][1], update=False )
+    lcd.update()
+            
+menu = [
+    [demo_medium_motor,    'Medium Motor'],
+    [demo_large_motor,     'Large Motor'],
+    [demo_move_steering,   'Move Steering'],
+    [demo_move_tank,       'Move Tank'],
+    [demo_status_light,    'Status Light'],
+    [demo_color_sensor,    'Color Sensor'],
+    [demo_ir_sensor,       'IR Sensor'],
+    [demo_motor_rotation,  'Motor Rotation']
+]
+
+# Start script
+
+robot.connect_motor( 'medium' )
+robot.connect_motor( 'left' )
+robot.connect_motor( 'right' )
+robot.connect_sensor( 'touch' )
+robot.connect_sensor( 'color' )
+robot.connect_sensor( 'infrared' )
+
+# Brickman uses virtual terminal 1 to control the brick display.
+# By changing to a different virtual terminal we can take over the
+# display even though brickman is still running.
+
+# Need advice on best way to avoid constantly being prompted for sudo password
+os.system("sudo chvt 7")
```

Comment:
If you run the script from the brickman menu (the `File system` one), then `chvt` call should not be necessary. But you can avoid a password prompt by changing `/etc/sudoers` file.